src/toffy/fov_watcher.py

```diff
@@ -198,12 +199,21 @@ def __init__(
                 callback to run over the entire run
             intermediate_callback (Callable[[None], None]):
                 run callback overriden to run on each fov
-            timeout (int):
+            fov_timeout (int):
                 number of seconds to wait for non-null filesize before raising an error
+            watcher_timeout (int):
+                length to wait for new file generation before timing out
         """
         super().__init__()
         self.run_folder = run_folder
 
+        self.last_event_time = datetime.now()
+        self.timer_thread = threading.Thread(
```

Comment:
I'm confused as to what the thread is up to here. Also does this thread get to be closed eventually?